packages/astro/src/astro-components/unstyled/SubscriptionDetailsButton.astro

```diff
@@ -0,0 +1,74 @@
+---
+import type { __experimental_SubscriptionDetailsButtonProps } from '@clerk/types';
+
+import type { HTMLTag, Polymorphic } from 'astro/types';
+import type { ButtonProps } from '../../types';
+import { addUnstyledAttributeToFirstTag, logAsPropUsageDeprecation } from './utils';
+
+type Props<Tag extends HTMLTag = 'button'> = Polymorphic<ButtonProps<Tag>> &
+  __experimental_SubscriptionDetailsButtonProps;
+
+import { generateSafeId } from '@clerk/astro/internal';
+
+const safeId = generateSafeId();
+
+if ('as' in Astro.props) {
+  logAsPropUsageDeprecation();
+}
+
+const {
+  as: Tag = 'button',
+  asChild,
+  for: _for,
+  subscriptionDetailsProps,
+  onSubscriptionCancel,
+  ...props
+} = Astro.props;
+
+const subscriptionDetailsOptions = {
+  for: _for,
+  onSubscriptionCancel,
+  ...subscriptionDetailsProps,
+};
+
+let htmlElement = '';
+
+if (asChild) {
+  htmlElement = await Astro.slots.render('default');
+  htmlElement = addUnstyledAttributeToFirstTag(htmlElement, safeId);
+}
+---
+
+{
+  asChild ? (
+    <Fragment set:html={htmlElement} />
+  ) : (
+    <Tag
+      {...props}
+      data-clerk-unstyled-id={safeId}
+    >
+      <slot>Subscription details</slot>
+    </Tag>
+  )
+}
+
+<script is:inline define:vars={{ props, subscriptionDetailsOptions, safeId }}>
+  const btn = document.querySelector(`[data-clerk-unstyled-id="${safeId}"]`);
+
+  btn.addEventListener('click', () => {
+    const clerk = window.Clerk;
+
+    // Authentication checks
+    if (!clerk.user) {
+      throw new Error('Ensure that `<SubscriptionDetailsButton />` is rendered inside a `<SignedIn />` component.');
+    }
+
+    if (!clerk.organization && subscriptionDetailsOptions.for === 'organization') {
+      throw new Error(
+        'Wrap `<SubscriptionDetailsButton for="organization" />` with a check for an active organization.',
+      );
+    }
+
+    return clerk.__internal_openSubscriptionDetails(subscriptionDetailsOptions);
```

Comment:
Since we can't pass a function as prop here, the `onSubscriptionCancel` prop is unusable, but we can use a custom event [like this](https://github.com/clerk/javascript/blob/f0fe516ed2e5d551090d680d973139bbc6408104/packages/astro/src/astro-components/interactive/UserButton/MenuItemRenderer.astro#L71,L74) 

```ts
const clickEvent = new CustomEvent('clerk:subscription-cancel');
clerk.__internal_openSubscriptionDetails({
  ...subscriptionDetailsOptions,
  onSubscriptionCancel: () => document.dispatchEvent(clickEvent),
});
```

and in the parent component, they can listen to it

```html
<script>
  document.addEventListener('clerk:subscription-cancel', (e) => {
     //...
  })
</script>
```